classes/author/Collector.inc.php

```diff
@@ -0,0 +1,232 @@
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * @copydoc CollectorInterface::getQueryBuilder()
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $q = DB::table('authors as a')
+            ->Join('publications as p', 'a.publication_id', '=', 'p.publication_id')
+            ->Join('submissions as s', 'p.submission_id', '=', 's.submission_id');
+
+        if (isset($this->contextIds)) {
+            $q->whereIn('s.context_id', $this->contextIds);
+        }
+
+        if (isset($this->familyName) && isset($this->givenName)) {
+            $familyName = $this->familyName;
+            $givenName = $this->givenName;
+            $q->leftJoin('author_settings as fn', 'a.author_id', '=', 'fn.author_id')
+                ->where(function ($q) use ($familyName) {
+                    $q->where('fn.setting_name', '=', 'familyName');
+                    $q->where('fn.setting_value', '=', $familyName);
+                });
+            $q->leftJoin('author_settings as gn', 'a.author_id', '=', 'gn.author_id')
+                ->where(function ($q) use ($givenName) {
+                    $q->where('gn.setting_name', '=', 'givenName');
+                    $q->where('gn.setting_value', '=', $givenName);
+                });
+        }
+
+        if (isset($this->publicationIds)) {
+            $q->whereIn('a.publication_id', $this->publicationIds);
+        }
+
+        if (isset($this->country)) {
+            $country = $this->country;
+            $q->join('author_settings as cs', 'a.author_id', '=', 'cs.author_id')
+                ->where(function ($q) use ($country) {
+                    $q->where('cs.setting_name', '=', 'country');
+                    $q->where('cs.setting_value', '=', $country);
+                });
+        }
+
+        if (isset($this->affiliation)) {
+            $affiliation = $this->affiliation;
+            $q->join('author_settings as afs', 'a.author_id', '=', 'afs.author_id')
+                ->where(function ($q) use ($affiliation) {
+                    $q->where('afs.setting_name', '=', 'affiliation');
+                    $q->where('afs.setting_value', '=', $affiliation);
+                });
+        }
+
+        if ($this->includeInBrowse) {
+            $q->where('a.include_in_browse', $this->includeInBrowse);
+        }
+
+        if (isset($this->count)) {
+            $q->limit($this->count);
+        }
+
+        if (isset($this->offset)) {
+            $q->offset($this->offset);
+        }
+
+        switch ($this->orderBy) {
+            case self::ORDERBY_SEQUENCE:
+                $q->orderBy('a.seq', 'asc');
+                break;
+            case self::ORDERBY_ID:
+            default:
+                $q->orderBy('a.author_id', 'asc');
+                break;
+        }
+
+        // Add app-specific query statements
+        HookRegistry::call('Author::Collector', [&$q, $this]);
+
+        $q->select(['*', 's.locale AS submission_locale']);
```

Comment:
I'd suggest moving this close to the top of the function, which would allow hook users to use `addSelect`. (Otherwise this function will clobber any additional columns hook registrants have added.)

Reply:
Also this!

Reply:
Moved it after first 2 joins